open-api/rest-catalog-open-api.yaml

```diff
@@ -3044,29 +3088,51 @@ components:
           type: array
           items:
             $ref: '#/components/schemas/PlanTask'
-        plan-status:
-            $ref: '#/components/schemas/PlanStatus'
 
+    EmptyResult:
+      type: object
+      description: Empty object result
 
-    RetrieveTasksResult:
+    PlanningResult:
```

Comment:
This is returned by both `GET` and `POST` to the `plan` endpoint. It contains either tasks (when status is "succeeded") or an error (when the status is "failed"). For submitted/cancelled, I added an empty result that's just above.